### PR TITLE
Add locale=user to REST API requests

### PR DESCRIPTION
**Pull request by TimothyBJacobs**
Trac ticket: https://core.trac.wordpress.org/ticket/51871

---
**This Pull Request is for code review only. Please keep all other discussion in the Trac ticket. Do not merge this Pull Request. See [GitHub Pull Requests for Code Review](https://make.wordpress.org/core/handbook/contribute/git/github-pull-requests-for-code-review/) in the Core Handbook for more details.**

### CODE REVIEW

File: src/js/_enqueues/wp/api-request.js [hunk 1]
```diff
@@ -55,18 +55,21 @@
...
-		if ( addNonceHeader ) {
-			for ( headerName in headers ) {
-				if ( headers.hasOwnProperty( headerName ) ) {
-					if ( headerName.toLowerCase() === 'x-wp-nonce' ) {
-						addNonceHeader = false;
-						break;
-					}
+		for ( headerName in headers ) {
+			if ( headers.hasOwnProperty( headerName ) ) {
+				// If an 'X-WP-Nonce' header (or any case-insensitive variation
```
Comment (adamsilverstein) [inline src/js/_enqueues/wp/api-request.js:64]: This section would read a little cleaner as a switch statement:
```js
switch( headerName.toLowerCase() ) {
	case 'x-wp-nonce':
		addNonceHeader = false;
		break;
	case 'accept':
		addAcceptHeader = false;
		break;
}

**Review by adamsilverstein — APPROVED** [review]
JS changes looks good @TimothyBJacobs! Left one minor feedback item.

File: src/js/_enqueues/wp/api-request.js (hunk 1 above)
Comment (TimothyBJacobs) [reply src/js/_enqueues/wp/api-request.js]: Fixed in a316b9fff86b4afc34753318de558fe46c399d14.

**Comment by TimothyBJacobs** [comment]
Thanks for the review @adamsilverstein! Pushed a commit to address that issue.

**Pull request closed without merging**